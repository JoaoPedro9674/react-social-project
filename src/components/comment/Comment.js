import React, { PropTypes } from 'react';
import moment from 'moment';

const Comment = (props) => {
  const { comment } = props;
  return (
    <div className="comment" key={comment.id}>
      <img
        src={comment.user.profilePicture}
        width={50}
        height={50}
        alt={comment.content}
      />
      <a>{comment.user.firstName} {comment.user.lastName}</a>
      <small>
        {moment(comment.date).fromNow()}
      </small>
      <p>
        {comment.content}
      </p>
    </div>
  );
};

Comment.propTypes = {
  comment: PropTypes.shape({
    content: PropTypes.string,
    user: PropTypes.object,
    date: PropTypes.string,
    likes: PropTypes.number,
  }),
};

export default Comment;
