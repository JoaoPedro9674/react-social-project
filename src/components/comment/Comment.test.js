import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import Comment from './Comment';

describe('<Comment/>', () => {
    let sandbox;
    beforeEach(() => {
        sandbox = sinon.sandbox.create();
    });
    afterEach(() => {
        sandbox.restore();
    });
    it('should render correctly', () => {
        const mockComment = {
            id: 'id',
            content: 'content',
            user: {
                firstName: 'mark',
                profilePicture: 'profilePicture',
                lastName: 'thomas'
            },
            date: new Date().toISOString(),
            likes: 10
        };
        const component = renderer.create(<Comment comment={mockComment} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
