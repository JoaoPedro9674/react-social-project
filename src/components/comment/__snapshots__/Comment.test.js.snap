exports[`<Comment/> should render correctly 1`] = `
<div
  className="comment">
  <img
    alt="content"
    height={50}
    src="profilePicture"
    width={50} />
  <a>
    mark
     
    thomas
  </a>
  <small
    className="date">
     
    a few seconds ago
     
  </small>
  <p>
     
    content
     
  </p>
</div>
`;
